main.go

```diff
@@ -0,0 +1,224 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+// ConfigSettings contains the necessary configuration data for the service
+type ConfigSettings struct {
+	Port  int      `yaml:"port"`
+	Repos []string `yaml:"repos"`
+}
+
+// NewConfigSettings creates a new ConfigSettings struct which manages service state
+func NewConfigSettings(configFile string) (*ConfigSettings, error) {
+	yamlData, err := ioutil.ReadFile(configFile)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var config ConfigSettings
+	err = yaml.Unmarshal(yamlData, &config)
+	if err != nil {
+		return nil, err
+	}
+
+	if config.Port == 0 {
+		log.Println("no port specified in the yaml config file, defaulting to 8000")
+		config.Port = 8000
+	}
+
+	if len(config.Repos) == 0 {
+		return nil, errors.New("no git repos were configured in the yaml config file, at least one repo must be listed for the service to query")
+	}
+
+	return &config, nil
+}
+
+func main() {
+	configFile := "config.yaml"
+	if len(os.Args) == 1 {
+		log.Println("no configuration file specified, defaulting to config.yaml")
+	}
+	if len(os.Args) > 1 {
+		configFile = os.Args[1]
+	}
+
+	serviceConfig, err := NewConfigSettings(configFile)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Register the search endpoint here
+	http.HandleFunc("/search", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Even better:
```suggestion
	http.HandleFunc("/search", serviceConfig.HandleSearch)
```
With `HandleSearch` being a method on the `ConfigSettings` struct:
```
func (config *ConfigSettings) HandleSearch(w http.ResponseWriter, r *http.Request)
```

Reply:
And if that was a method on that struct, the struct should be named something different like `Manager` or something that signifies that once initialized, it is the thing that will manage the state and functionality of the config and handlers.